src/arrayMethodReduce.js

```diff
@@ -4,8 +4,37 @@
  * Implement method Reduce
  */
 function applyCustomReduce() {
-  [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+  [].__proto__.reduce2 = function(callback, initialValue = this[0]) {
+    let previous = this[0];
+    let current;
+
+    function loopFunc(arr) {
+      for (let i = 0; i < arr.length; i++) {
+        current = arr[i];
+        previous = callback(previous, current, i, arr);
+      }
+    }
+
+    function loopFuncOne(arr) {
+      for (let i = 1; i < arr.length; i++) {
+        current = arr[i];
+        previous = callback(previous, current, i, arr);
+      }
+    }
```

Comment:
You still have almost identical functions, you can make it more concise. Only one parameter is changing.